R-package/R/utils.R

```diff
@@ -0,0 +1,840 @@
+#' #                                      sparse_max_features = 80000,
+#' #                                      sparse_min_occurences = 5,
+#' #                                      calc_prob="sigmoid", n_jobs = 1,
+#' #                                      verbose = 0)
+#' @examples
+#'
+#' if (reticulate::py_available() && reticulate::py_module_available("rgf.sklearn")) {
+#'
+#'   library(RGF)
+#'
+#'   set.seed(1)
+#'   x = matrix(runif(100000), nrow = 100, ncol = 1000)
+#'
+#'   y = sample(1:2, 100, replace = TRUE)
+#'
+#'   fast_RGF_class = FastRGF_Classifier$new(max_leaf = 50)
+#'
+#'   fast_RGF_class$fit(x, y)
+#'
+#'   preds = fast_RGF_class$predict_proba(x)
+#' }
+
+FastRGF_Classifier <- R6::R6Class("FastRGF_Classifier",
+
+                                  inherit = Internal_class,
+                                  
+                                  lock_objects = FALSE,
+
+                                  public = list(
+
+                                    initialize = function(n_estimators = 500, max_depth = 6, max_leaf = 50, tree_gain_ratio = 1.0, min_samples_leaf = 5, loss = "LS", l1 = 1.0,
+                                                         l2 = 1000.0, opt_algorithm = "rgf", learning_rate = 0.001, max_bin = NULL, min_child_weight = 5.0,
+                                                         data_l2 = 2.0, sparse_max_features = 80000, sparse_min_occurences = 5, calc_prob="sigmoid", n_jobs = 1, verbose = 0) {
+
+                                      # exceptions for 'min_samples_leaf', 'max_bin'
+                                      #---------------------------------------------
+
+                                      if (min_samples_leaf >= 1.0) {                               # must be either > as.integer(1.0) or in (0, 0.5]
+
+                                        min_samples_leaf = as.integer(min_samples_leaf)
+                                      }
+
+                                      if (!is.null(max_bin)) {                                     # must be either NULL or an integer
+
+                                        max_bin = as.integer(max_bin)
+                                      }
+
+
+                                      # initialize FastRGF_Regressor
+                                      #-----------------------------
+
+                                      private$rgf_init = RGF_mod$FastRGFClassifier(as.integer(n_estimators), as.integer(max_depth), as.integer(max_leaf), tree_gain_ratio, min_samples_leaf,
+                                                                                   loss, l1, l2, opt_algorithm, learning_rate, max_bin, min_child_weight, data_l2, as.integer(sparse_max_features),
+                                                                                   as.integer(sparse_min_occurences), calc_prob, as.integer(n_jobs), as.integer(verbose))
+                                    }
+                                  )
+)
```

Comment:
1 blank line at the end of the file
![image](https://user-images.githubusercontent.com/25141164/43592944-504ab852-967f-11e8-8fb9-1a55a8789c71.png)